src/pages/design/[sectionId]/[articleId].tsx

```diff
@@ -9,18 +9,64 @@ import {DesignLayout} from '../../../components/DesignLayout/DesignLayout';
 import {Layout} from '../../../components/Layout/Layout';
 import {sections as designSections} from '../../../content/design';
 import {getI18nProps} from '../../../utils/i18next';
+import {type MetaProps, getDesignArticleMeta} from '../../../utils/lib';
 
 export const getServerSideProps: GetServerSideProps = async (ctx) => {
+    const sectionId = ctx.params?.sectionId as string;
+    const articleId = ctx.params?.articleId as string;
+    const locale = ctx.locale || 'en';
+
+    // Find the section and article
+    const section = designSections.find((item) => item.id === sectionId);
+    const article = section?.articles.find((item) => item.id === articleId);
+
+    if (!section || !article) {
+        return {
+            notFound: true,
+        };
+    }
+
+    // Get i18n props to access translations
+    const i18nProps = await getI18nProps(ctx, ['design-article', 'design-articles-info']);
+
+    // Create a temporary t function for title extraction
+    const translations =
+        i18nProps._nextI18Next?.initialI18nStore?.[locale]?.['design-articles-info'] || {};
+    const getTitle = (key: string) => translations[key] || key;
+
+    // Get localized titles
+    const sectionTitle = getTitle(`section_${sectionId}_title`);
+    const articleTitle = getTitle(`section_${sectionId}_article_${articleId}_title`);
+
+    // Generate meta description from static config
+    const articleMeta = await getDesignArticleMeta(
+        sectionId,
+        sectionTitle,
+        articleId,
+        articleTitle,
+        locale,
+        `${articleTitle} design guide from Gravity UI`,
+    );
+
     return {
         props: {
-            sectionId: ctx.params?.sectionId,
-            articleId: ctx.params?.articleId,
-            ...(await getI18nProps(ctx, ['design-article', 'design-articles-info'])),
+            sectionId,
+            articleId,
+            articleMeta,
+            ...i18nProps,
         },
     };
 };
 
-export const ArticlePage = ({sectionId, articleId}: {sectionId: string; articleId: string}) => {
+export const ArticlePage = ({
```

Comment:
Remove this code in `ArticlePage` plz
```ts
if (!section || !article) {
    return <Error statusCode={404} />;
}
```

now we have a more correct condition in `getServerSideProps`:
```
if (!section || !article) {
    return {
        notFound: true,
    };
}
```